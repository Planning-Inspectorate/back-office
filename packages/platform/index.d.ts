import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
