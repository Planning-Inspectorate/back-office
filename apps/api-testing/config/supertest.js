// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
