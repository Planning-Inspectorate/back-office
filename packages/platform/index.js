export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
