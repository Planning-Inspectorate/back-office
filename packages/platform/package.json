{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
