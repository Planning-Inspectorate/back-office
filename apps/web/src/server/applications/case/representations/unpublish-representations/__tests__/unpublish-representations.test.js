import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const baseUrl = '/applications-service/case/1/relevant-representations/unpublish-representations';

const mockCaseData = { title: 'Unpublish relevant representations' };
const mockRepresentations = {
	items: [
		{ id: '1', status: 'PUBLISHED' },
		{ id: '2', status: 'DRAFT' }
	]
};

describe('unpublish-representations', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => nock.cleanAll());

	describe('GET /applications-service/:caseId/unpublish-representations/unpublish-representations', () => {
		describe('It renders the unpublish page', () => {
			const nocks = () => {
				nock('http://test/').get('/applications/1').reply(200, mockCaseData);
				nock('http://test/').get('/applications/1').reply(200, mockCaseData);
				nock('http://test/').get('/applications/1/representations').reply(200, mockRepresentations);
			};
			nocks();

			it('should render the page with the publish button', async () => {
				const response = await request.get(baseUrl);

				const headingElement = parseHtml(response.text, { rootElement: '.govuk-heading-l' });
				const buttonElement = parseHtml(response.text, {
					rootElement: 'form button[data-module="govuk-button"]'
				});
				expect(headingElement.innerHTML).toContain('Unpublish relevant representations');
				expect(buttonElement.innerHTML).toContain('Unpublish all representations');
			});
		});
	});

	describe('POST /applications-service/case/:caseId/relevant-representations/unpublish-representations', () => {
		// const publishedReps = [
		// 	{
		// 		id: 1,
		// 		status: 'PUBLISHED',
		// 		reference: 'mock-reference-1',
		// 		redacted: true,
		// 		received: '2023-10-10T10:47:21.846Z',
		// 		firstName: 'mock name',
		// 		lastName: 'mock last name',
		// 		organisationName: 'mock org'
		// 	},
		// 	{
		// 		id: 2,
		// 		status: 'PUBLISHED',
		// 		reference: 'mock-reference-2',
		// 		redacted: true,
		// 		received: '2023-10-10T10:47:21.846Z',
		// 		firstName: 'mock name',
		// 		lastName: 'mock last name',
		// 		organisationName: 'mock org'
		// 	}
		// ];
		// const allReps = [...publishedReps, {
		// 	id: 3,
		// 	status: 'VALID',
		// 	reference: 'mock-reference-3',
		// 	redacted: true,
		// 	received: '2023-10-10T10:47:21.846Z',
		// 	firstName: 'mock name',
		// 	lastName: 'mock last name',
		// 	organisationName: 'mock org'
		// }];
		const validReps = [
			{
				id: 1,
				status: 'VALID',
				reference: 'mock-reference-1',
				redacted: true,
				received: '2023-10-10T10:47:21.846Z',
				firstName: 'mock name',
				lastName: 'mock last name',
				organisationName: 'mock org'
			},
			{
				id: 2,
				status: 'VALID',
				reference: 'mock-reference-2',
				redacted: true,
				received: '2023-10-10T10:47:21.846Z',
				firstName: 'mock name',
				lastName: 'mock last name',
				organisationName: 'mock org'
			}
		];

		// describe('when there are published representations', () => {
		// 	beforeEach(() => {
		// 		nock('http://test/').get('/applications/1').reply(200, mockCaseData);
		// 		nock('http://test/').get('/applications/1/representations').reply(200, { items: allReps });
		// 		nock('http://test/').get('/applications/1').reply(200, mockCaseData);
		// 		nock('http://test/').get('/applications/1').reply(200, mockCaseData);
		// 		nock('http://test/').get('/applications/1/representations').reply(200, { items: allReps });
		// 		nock('http://test/')
		// 			.get('/applications/1/representations/publishable')
		// 			.reply(200, { publishedRepIds: [1, 2] });
		// 	});
		//
		// 	it('should redirect with unpublished param if published representations exist', async () => {
		// 		let response;
		// 		try {
		// 			response = await request
		// 				.post('/applications-service/case/1/relevant-representations/unpublish-representations')
		// 				.send();
		// 		} catch (err) {
		// 			console.error('Test error:', err);
		// 			throw err;
		// 		}
		// 		console.log('<><><><><><>  Response <><><><><><> ', { response });
		// 		expect(response.status).toBe(302);
		// 		expect(response.headers.location).toBe('/applications-service/case/1/relevant-representations?unpublished=2');
		// 	});
		// });

		describe('when there are no published representations', () => {
			beforeEach(() => {
				nock('http://test/').get('/applications/1').reply(200, mockCaseData);
				nock('http://test/')
					.get('/applications/1/representations')
					.reply(200, { items: validReps });
				nock('http://test/')
					.patch('/applications/1/representations/publishable')
					.reply(200, { publishedRepIds: [] });
			});

			it('should redirect without unpublished param if no published representations', async () => {
				const response = await request
					.post('/applications-service/case/1/relevant-representations/unpublish-representations')
					.send();
				expect(response.status).toBe(302);
				expect(response.headers.location).toBe(
					'/applications-service/case/1/relevant-representations'
				);
			});
		});
	});
});
