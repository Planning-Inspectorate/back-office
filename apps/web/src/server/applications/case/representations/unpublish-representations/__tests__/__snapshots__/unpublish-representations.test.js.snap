// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unpublish-representations GET /applications-service/:caseId/relevant-representations/unpublish-representations and the unpublishable rep count is > 0 should render the page with the unpublish button 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row unpublish-representations">
        <div class="govuk-grid-column-two-thirds">
            <h1 class="govuk-heading-l">Unpublish relevant representations</h1>
            <p class="govuk-body"><strong>Project Name:</strong>&nbsp;mock title</p>
            <p class="govuk-body">There are<strong class="publishedRepsCount">3</strong> published representations.</p>
            <p             class="govuk-body">Unpublishing will remove these representations from the external website.</p>
                <div                 class="govuk-inset-text">You can republish a representation by changing its status to "Valid" and
                    selecting "Publish representations" from the publishing queue.</div>
        <form         method="post">
            <button type="submit" data-prevent-double-click="true" class="govuk-button govuk-button--warning"
            data-module="govuk-button">Unpublish all representations</button>
            </form>
    </div>
    </div>
</main>"
`;

exports[`unpublish-representations GET /applications-service/:caseId/relevant-representations/unpublish-representations and the unpublishable rep count is 0 should render the page without the unpublish button 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <h1 class="govuk-heading-xl">Sorry, there is a problem with the service</h1>
            <p class="govuk-body">If the problem persists, <a href="https://intranet.planninginspectorate.gov.uk/task/report-it-problem/"
                class="govuk-link">report an IT problem or make an IT request.</a>
            </p>
        </div>
    </div>
</main>"
`;
